services/softlayer_network_storage.go

```
@@ -77,8 +90,9 @@ func (slns *softLayer_Network_Storage_Service) CreateIscsiVolume(size int, locat
 }
 
 func (slns *softLayer_Network_Storage_Service) DeleteIscsiVolume(volumeId int, immediateCancellationFlag bool) error {
+	ObjectFilter := string(`{"iscsiNetworkStorage":{"id":{"operation":` + strconv.Itoa(volumeId) + `}}}`)
 	accountService, _ := slns.client.GetSoftLayer_Account_Service()
-	iscsiStorages, _ := accountService.GetIscsiNetworkStorage()
+	iscsiStorages, _ := accountService.GetIscsiNetworkStorageWithFilter(ObjectFilter)
 
```

Same deal here again. We cannot ignore errors.